apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -511,7 +543,11 @@ export default NextAuth({
               where: { email: existingUserWithEmail.email },
               data: { password: null },
             });
-            return true;
+            if (existingUserWithEmail.twoFactorEnabled) {
+              return `/auth/login?totp=${await signJwt({ email: existingUserWithEmail.email })}`;
```

Comment:
This is the redirection needed to let the user input a totp code in an external provider use case.